src/main/webapp/app/localci/build-agents/build-agents.component.ts

```diff
@@ -56,21 +62,35 @@ export class BuildAgentsComponent implements OnInit, OnDestroy {
     load() {
         this.restSubscription = this.buildAgentsService.getBuildAgents().subscribe((buildAgents) => {
             this.buildAgents = buildAgents;
-            this.setBuildAgentBuildJobIds(buildAgents);
+            this.setRecentBuildJobsDuration(buildAgents);
         });
     }
 
-    /**
-     * This method is used to set the build job ids string for each build agent.
-     * @param buildAgents the build agents for which the build job ids string should be set
-     */
-    setBuildAgentBuildJobIds(buildAgents: BuildAgent[]) {
+    setRecentBuildJobsDuration(buildAgents: BuildAgent[]) {
         for (const buildAgent of buildAgents) {
-            if (buildAgent.runningBuildJobs) {
-                buildAgent.runningBuildJobsIds = buildAgent.runningBuildJobs.map((buildJob: BuildJob) => buildJob.id).join(', ');
-            } else {
-                buildAgent.runningBuildJobsIds = '';
+            const recentBuildJobs = buildAgent.recentBuildJobs;
+            if (recentBuildJobs) {
+                for (const buildJob of recentBuildJobs) {
+                    if (buildJob.jobTimingInfo?.buildStartDate && buildJob.jobTimingInfo?.buildCompletionDate) {
+                        const start = dayjs(buildJob.jobTimingInfo.buildStartDate);
+                        const end = dayjs(buildJob.jobTimingInfo.buildCompletionDate);
+                        buildJob.jobTimingInfo.buildDuration = end.diff(start, 'milliseconds') / 1000;
+                    }
+                }
             }
         }
     }
+
+    cancelBuildJob(buildJobId: string) {
+        this.buildQueueService.cancelBuildJob(buildJobId).subscribe();
+    }
+
+    cancelAllBuildJobs(buildAgentName: string) {
+        const buildAgent = this.buildAgents.find((agent) => agent.name === buildAgentName);
+        if (buildAgent && buildAgent.name) {
+            this.buildQueueService.cancelAllRunningBuildJobsForAgent(buildAgent.name).subscribe();
+        }
+    }
+
+    protected readonly TriggeredByPushTo = TriggeredByPushTo;
```

Comment:
This should be moved to top, where the other attributes are placed which are used in the html